src/bootstrap/tool.rs

```diff
@@ -736,6 +738,8 @@ macro_rules! tool_extended {
 
 // Note: tools need to be also added to `Builder::get_step_descriptions` in `builder.rs`
 // to make `./x.py build <tool>` work.
+// Note: Most submodule updates for tools are handled by bootstrap.py, since they're needed just to
+// build bootstrap.
```

Comment:
Can we rephrase this to something like "invoke Cargo"? In particular, no submodules are *actually* required to build bootstrap, and we could avoid the connection to bootstrap by creating a separate workspace for it. But, that wouldn't help with the need for these to get init'd anyway since they're still needed for any cargo invocation pretty much.

Reply:
> But, that wouldn't help with the need for these to get init'd anyway since they're still needed for any cargo invocation pretty much.

We could avoid cloning most of the tool subdirectories if we kept the compiler in a separate workspace from the standard library. cc https://github.com/rust-lang/rust/pull/76533

Anyway, I updated the comment.